File: Valid Palindrome2.java
class Solution {
  public boolean isPalindrome(String s) {
    // TODO: Write your code here
    char[] strArray = s.toCharArray();
    int left = 0;
    int right = strArray.length - 1;

    while(left < right){
      while(left < right && !Character.isLetterOrDigit(strArray[left])){
        left++;
      }

      while(left < right && !Character.isLetterOrDigit(strArray[right])){
        right--;
      }

      if(Character.toLowerCase(strArray[left]) != Character.toLowerCase(strArray[right])){
        return false;
      }
      left++;
      right--;
    }

    return true; 
  }
}

File: Valid palindrome 2.py
class Solution:
    def isPalindromePossible(self, s: str) -> bool:
        # ToDo: Write Your Code Here.
        if len(s) == 0:
            return False
        left = 0
        right = len(s) - 1

        while left < right:
            if s[left] != s[right]:
                return self.isPalindrome(s, left + 1, right) or self.isPalindrome(s, left, right -1)
            left += 1
            right -= 1
        return True

    
    def isPalindrome(self, s: str, left: int, right: int) -> bool:
        while left < right:
            if s[left] != s[right]:
                return False
            left += 1
            right -= 1
        return True

